src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -161,6 +160,41 @@ private async Task<ObservedDataMapping> copyObservedData(ObservedDataMapping obs
          };
       }
 
+      private Task<Section[]> copySectionContents(Section[] sections)
+      {
+         if (sections == null)
+            return Task.FromResult(Array.Empty<Section>());
+
+         return Task.WhenAll(sections.Select(copySectionContent));
+      }
+
+      private async Task<Section> copySectionContent(Section section)
+      {
+         Task<string> downloadRemoteSectionContent() => downloadRemoteFile(section.Content, CONTENT_DOWNLOAD_FOLDER, "Content");

```

Comment:
In theory we could alwyas make sure that remote file are downloaded to a unique name (e.g. if the file exists, just createa  new one)